core/src/ast/expr_decoder.rs

```diff
@@ -0,0 +1,313 @@
+use super::{Aggregate, AstLiteral, CountArgExpr, Expr};
+
+pub fn decode(expr: &Expr) -> String {
+    match expr {
+        Expr::Identifier(s) => s.to_string(),
+        Expr::BinaryOp { left, op, right } => {
+            format!("{:} {:} {:}", decode(&*left), op, decode(&*right))
```

Comment:
```rust
format!("{:} {:} {:}",
```
Here `{:}` looks unnecessary, when you expect to use `Display` trait, then `{}` is enough

Reply:
> format!("{}", decode(&*left), op, decode(&*right))

Note that `{}` takes only one parameter.
In the code you wrote above, there is only one parentheses, but there are 3 parameters.